src/lib/utils.ts

```diff
@@ -71,10 +71,13 @@ export const columnErrorQueryBuilder = (columns: string[]) => {
 }
 
 export const filterQueryBuilder = (filter: string) => {
-  // SQL
-  // return `select * from table where (${filter})`
-  // PRQL
-  return filter ? `filter ${filter}` : ""
+  if (filter && filter.includes("text.lower") && filter.includes("not")) {
+    // If the filter includes both "text.lower" and "not"
+    return `filter !${filter.replace("not","")}`;
```

Comment:
this is introducing a bug, it is the wrong place to add this transformation, should be handled in FilterBlock

Reply:
initially filterQueryBuilder was like this 
export const filterQueryBuilder = (filter: string) => {
    return filter ? `filter ${filter}` : "";
}

to avoid case sensitivity i decided to lowercase both column value and valuestr 

so i have used inbuilt method text.lower to convert column value to lowercase 
![image](https://github.com/pretzelai/pretzelai/assets/113238477/e6375b0c-b5dd-4ba9-b12a-f8f7f9e25bc8)


 operatorStr = "| text.lower | text.contains"
        valueStr=valueStr.toLowerCase()

but the issue came with  doesn't contain final sql query  generated was "SELECT
  *
FROM
  "InputTable"
WHERE
  LOWER(NOT "Name") LIKE CONCAT('%', 'acode', '%')
LIMIT
  10000
"
to overcome i decided to change the query only for "not contains"

 if (filter && filter.includes("text.lower") && filter.includes("not")) {
    // If the filter includes both "text.lower" and "not"
    return `filter !${filter.replace("not","")}`;
  }
this condition check whether filter includes "textlower" and includes "not" i have added this condition  `filter !${filter.replace("not","")}`;
this resulted in correct sql query 
"SELECT
  *
FROM
  "InputTable"
WHERE
  NOT LOWER("Name") LIKE CONCAT('%', 'acode', '%')
LIMIT
  10000
"

and the issue solved finally